src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs

```diff
@@ -20,17 +23,68 @@ public interface IHttpClientFactory
         /// Creating a new <see cref="HttpClient"/> consumes one free socket which may not be released
         /// by the Operating System until sometime after the client is disposed, leading to possible free
         /// socket exhaustion.
+        /// <para/>
+        /// The client instance is configured for use behind a proxy using Git's http.proxy configuration
+        /// for the local repository (the current working directory), if found.
         /// </remarks>
         /// <returns>New client instance with default headers.</returns>
         HttpClient CreateClient();
     }
 
     public class HttpClientFactory : IHttpClientFactory
     {
+        private readonly ITrace _trace;
+        private readonly ISettings _settings;
+        private readonly IStandardStreams _streams;
+
+        public HttpClientFactory(ITrace trace, ISettings settings, IStandardStreams streams)
+        {
+            EnsureArgument.NotNull(trace, nameof(trace));
+            EnsureArgument.NotNull(settings, nameof(settings));
+            EnsureArgument.NotNull(streams, nameof(streams));
+
+            _trace = trace;
+            _settings = settings;
+            _streams = streams;
+        }
+
         public HttpClient CreateClient()
         {
-            // Initialize a new HttpClient
-            var client = new HttpClient();
+            _trace.WriteLine("Creating new HTTP client instance...");
+
+            HttpClientHandler handler;
+
+            if (TryCreateProxy(out IWebProxy proxy))
+            {
+                _trace.WriteLine("HTTP client is using the configured proxy.");
```

Comment:
To verify do we print out the details of the proxy being using anywhere?  That would be helpful for debugging i think.

Reply:
Yes, we print out the proxy settings used each time a new `HttpClient` instance is created.

Reply:
https://github.com/mjcheetham/Git-Credential-Manager-Core/blob/28d9cb68dacfb724414ff876b391e68e2d9792c1/src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs#L145